File: src/java/org/javaturk/dp/ch03/ocp/factory/solution/DirectorFactory.java
package org.javaturk.dp.ch03.ocp.factory.solution;

import org.javaturk.dp.ch04.factoryMethod.employee.EmployeeRandomizer;

public class DirectorFactory implements Factory {

	@Override
	public Director create() {
		Director director = new Director(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
				EmployeeRandomizer.createYear(), "Management", "Management", 5000);
		return director;
	}

}

File: src/c#/Creational/FactoryMethod/FactoryMethod1/DirectorFactory.cs
using System;
using Util;

namespace FactoryMethod1
{
    public class DirectorFactory : IFactory
    {
        public Employee Create()
        {
            Director director = new Director(EmployeeRandomizer.CreateId(), EmployeeRandomizer.CreateName(),
                    EmployeeRandomizer.CreateYear(), "Management", "Management", 5000);
            return director;
        }
    }
}
